lock-api/src/main/java/com/palantir/lock/client/TransactionStarter.java

```diff
@@ -53,20 +53,16 @@
     private final DisruptorAutobatcher<Void, StartIdentifiedAtlasDbTransactionResponse> autobatcher;
     private final LockLeaseService lockLeaseService;
 
-    private TransactionStarter(
-            DisruptorAutobatcher<Void, StartIdentifiedAtlasDbTransactionResponse> autobatcher,
-            LockLeaseService lockLeaseService) {
-        this.autobatcher = autobatcher;
+    private TransactionStarter(LockLeaseService lockLeaseService, LockWatchEventCache lockWatchEventCache) {
+        this.autobatcher = Autobatchers
+                .independent(consumer(lockLeaseService, lockWatchEventCache))
+                .safeLoggablePurpose("transaction-starter")
+                .build();
         this.lockLeaseService = lockLeaseService;
     }
 
     static TransactionStarter create(LockLeaseService lockLeaseService, LockWatchEventCache lockWatchEventCache) {
-        DisruptorAutobatcher<Void, StartIdentifiedAtlasDbTransactionResponse> autobatcher = Autobatchers
-                .independent(consumer(lockLeaseService, lockWatchEventCache))
-                .safeLoggablePurpose("transaction-starter")
-                .build();
-        return new TransactionStarter(autobatcher,
-                lockLeaseService);
```

Comment:
I don't know what doing it this way gave us, but collapsing into the constructor allows us to make the consumer non-static so that it can call the unlock method.

Reply:
The general idea here is that through much of Atlas (at least recently written parts), constructors don't generally create new resources: normally this allows better use of e.g. mocks or stubs in tests, though admittedly less so in this case. I'd still prefer this because conventionally `create` alerts the user that there is some not-completely-trivial construction going on, which there is in this case with the disruptor.

I'd suggest we keep making the consumer in a static way, though without changing the public API of `TransactionStarter`: if we extract a private static method that does what `unlock` does with a passed-in `LockLeaseService` (e.g. `determineAndExecuteUnlocks()`) it should be simple to call that from `getStartTransactionResponses`, and `unlock` can call that as well.

Reply:
Interesting - that gives some context on why it was written this way (whereas my instinct is to use a constructor).